models/actions/task.go

```diff
@@ -220,9 +220,8 @@ func CreateTaskForRunner(ctx context.Context, runner *ActionRunner) (*ActionTask
 		return nil, false, err
 	}
 	defer commiter.Close()
-	ctx = dbCtx.WithContext(ctx)
 
-	e := db.GetEngine(ctx)
+	e := db.GetEngine(dbCtx)
```

Comment:
That's what I have suggested before: always use "ctx, commiter, err := db.TxContext(ctx)`, do not introduce any new `dbCtx`, it could avoid a lot of low-level mistakes.